FunctionalPreprocessing/FunctionalPreprocessing.XNAT_MARK_RUNNING_STATUS

```diff
@@ -77,14 +93,22 @@ get_options()
 				g_subject=${argument#*=}
 				index=$(( index + 1 ))
 				;;
-			--classifier=*)
-				g_classifier=${argument#*=}
-				index=$(( index + 1 ))
-				;;
+            --session=*)
+                g_session=${argument/*=/""}
+                index=$(( index + 1 ))
+                ;;
 			--scan=*)
 				g_scan=${argument#*=}
 				index=$(( index + 1 ))
 				;;
+            --resource=*)
+                g_resource=${argument/*=/""}
+                index=$(( index + 1 ))
+                ;;
+            --reason=*)
```

Comment:
I don't think you should let the user of this script specify the `--reason=` parameter. I think you should build the reason string inside this code depending upon which of the `--submitted`, `--queued`, `--running`, `--not-running`, `--not-queued` or `--done` parameters are specified.

So in the `get_options` function, you can still set a value for the `g_reason` variable. Just set its value based on which of the above options were specified. You can do this in the case statement.

So if `--submitted` is specified, then set `g_running="TRUE"`, but also set `g_reason="submitted"`. If `--queued` is specified, then set `g_running="TRUE"`, but also set `g_reason="running"`. Etc.

Reply:
added